src/ucp/rma/amo_sw.c

```diff
@@ -202,11 +202,12 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_atomic_req_handler, (arg, data, length, am_fl
 {
     ucp_atomic_req_hdr_t *atomicreqh = data;
     ucp_worker_h worker              = arg;
-    ucp_ep_h ep                      = ucp_worker_get_ep_by_id(worker,
-                                                        atomicreqh->req.ep_id);
     ucp_rsc_index_t amo_rsc_idx      = ucs_ffs64_safe(worker->atomic_tls);
     ucp_request_t *req;
+    ucp_ep_h ep;
 
+    ep = UCP_WORKER_GET_EP_BY_ID(worker, atomicreqh->req.ep_id, 0,
```

Comment:
why no check for closed ep?

Reply:
we could do the local atomic operation

Reply:
then why check for ep? Imo, always dropping the message/request if ep is closing/failing, would be more consistent.
Also you could do local atomic even if no ep found as well